ibmvmc/ibmvmc.c

```diff
@@ -1743,6 +1808,10 @@ static int ibmvmc_probe(struct vio_dev *vdev, const struct vio_device_id *id)
 	dev_set_drvdata(&vdev->dev, adapter);
 
 	return 0;
+
+crq_failed:
+	kthread_stop(adapter->reset_task);
+	return -1;
```

Comment:
Sorry, just do the NULL from above here, you don't need to return the PTR_ERR. But change -1 to EPERM. I don't understand why we override the rc's in a lot of these cases.